### DISCUSSION TITLE
Qual Framework CSS estão usando?

### DISCUSSION THREAD

**Discussion by victorserpac** [frontendbr/forum#16] [Perguntas e Respostas]
Estou em dúvida entre as várias opções de Framework para usar em um novo projeto e gostaria de saber a opinião de vocês sobre qual está usando no momento e porquê. Ou se não usa, queria saber a opinião também pelo motivo.

Vejo que o mais popular é obviamente o [Bootstrap](http://getbootstrap.com), mas também tenho visto algumas alternativas interessantes como [Foundation](http://foundation.zurb.com), [Semantic UI](http://semantic-ui.com), [UIkit](http://getuikit.com) entre outros e gostaria de saber as experiências de vocês.

**Comment by tcelestino** [comment]
Quando fazia freelas, nunca usava frameworks, pois queria ter o "controle" da interface de cada projeto, mas também porque a grande maioria dos designers que fiz trabalhos, não conheciam ou tinha preguiça de pesquisar sobre frameworks/grids etc...

Estou com um projeto, que pretendo usar o [Materialize](http://materializecss.com/about.html).

**Comment by felipefialho** [comment]
Carario, tinha escrito um texto enorme e a página atualizou, vamos lá de novo.

Sempre fui um grande fã do Bootstrap. Foi muito importante para eu definir e aprender questões de código componentizado, documentação, entre outras coisas que o Bootstrap oferece.

Inclusive fiquei muito contente quando tive um pull request aceito no projeto.

Hoje em dia meus projetos e meu workflow já estão bem diferentes, então só utilizo o Bootstrap para CMS simples, sem tratamento especial, os componentes são uma mão na roda para os  back ends utilizarem rapidamente.

Porém quero ressaltar que continuo usando o Carrossel do Bootstrap. Até hoje não encontrei nada melhor do que ele e não tenho tempo nem saco para fazer um na mão (é o único plugin desses básicos que utilizo solução de terceiros e não faço na mão).  O Carrossel é leve, flexível, escalável e fácilmente customizável para qualquer projeto.

Como hoje em dia estou utilizando o Zepto na maioria dos projetos, encontrei uma solução muito interessante do Bootstrap produzido com JavaScript nativo. Então utilizo esse componente sem maiores problemas.

Segue link http://thednp.github.io/bootstrap.native/

**Comment by yurimatheus** [comment]
Victor a minha sugestão é a seguinte, use:

Bootstrap - Para criar sites mais simples com poucas personalizações;
Foundation - Caso queira trabalhar com um Responsivo melhor por causa dos
seus grids inteligentes;
Semantic UI - Caso deseje algo bonito, realmente bonito! Poucos usam o
Semantic UI, eu mesmo dei uma estudada e criei alguns templates... espero
ter mais tempo para criar outros. Depois dê uma olhada no meu repo:
https://github.com/yurimatheus/semantic-ui_templates

Sent with MailTrack
https://mailtrack.io/install?source=signature&lang=en&referral=y.matheusdesign@gmail.com&idSignature=22

_Cordialmente,_

_Yuri Gomes._

Em 2 de fevereiro de 2016 19:51, Luiz Felipe Tartarotti Fialho <
notifications@github.com> escreveu:

**Comment by raphaklaus** [comment]
Já usei muito o Bootstrap, mas ele é bem pesado, tem algumas dependências que não concordo, e particularmente não gosto muito de trabalhar com responsividade no sistema de grids dele. 

Ultimamente estou meio chato com performance e também estou bem minimalista :stuck_out_tongue: 

CSS: [Miligram](https://milligram.github.io/) (apenas 2 kb gzipado)

Javascript: [ZeptoJS](http://zeptojs.com/) (apenas 9.1 kb gzipado)

Ícones: Não uso mais packs como font-awesome (que tem 12327348723 ícones que ninguém usa). Procuro pelas SVGs (só buscar no Google: "Free SVG Icons") que meu projeto precisará e incluo. Se bobiar uma landing page, por exemplo, ganha brincando +40kb só por não ter isso.

Fontes: Priorizo .woff2 nos browsers que suportam por ser um formato mais leve.

O principal: Eu não gosto de usar frameworks de UI porque acho que eles pesam mais e torna o projeto "igual a todos" e sem personalidade.

**Comment by kazzkiq** [comment]
**Projetos pequenos:**
Miligram/Grid Bootstrap
Sprint.js

**Projetos médios/grades:**
Pure.css ou Bootstrap
Sprint.js
Riot.js (Sim, RIOT, não confundir com React)

---

[**Miligram**](http://milligram.github.io/), [**Bootstrap**](http://getbootstrap.com/) e [**Pure**](http://purecss.io/) dispensam apresentações.

[**Sprint.js**](https://github.com/bendc/sprint) é uma biblioteca com basicamente a mesma API do jQuery, mas focada em performance e em navegadores mais novos (imagine um jQuery mais enxuto e muito mais performático). Não é pau-pra-toda-obra igual o jQuery, mas eu diria que atende 90% dos casos.

[**Riot.js** ](http://riotjs.com/) é uma biblioteca para criação de componentes semelhante ao React ou Polymer, mas bem mais leve e com uma sintaxe mil vezes mais intuitiva (sério, depois que você escreve um componente nela, não quer mais largar). A desvantagem é que ela pode ser simples até demais, e se você busca uma biblioteca com padrões de desenvolvimento mais bem definidos (como React, Ember ou Angular), ela pode não agradar.

Fora isso eu geralmente utilizo LESS ou SASS para pré-processamento CSS, e Gulp como task-runner.

**Comment by felipefialho** [comment]
@raphaklaus, você tocou em um ponto importante.

Muito cuidado ao usar tudo de um framework ou pacote de ícones. Provavelmente você não vai usar nem 30% disso, faça uma limpa e compile apenas o que realmente está usando.

**Comment by mustbetommy** [comment]
Eu não curto muito usar Frameworks. A única coisa que uso é a grid do Bootstrap porque já me acostumei, mas estou pensando em mudar. E para ícones, uso o Flaticon, um site onde você monta um pack de ícones único e baixa como icofont, png, jpg e até psd sem precisar pagar nada.

**Comment by raphaklaus** [comment]
@kazzkiq legal! O Sprint.js eu não conhecia. Pelos testes parece ser mais leve e rápido que o ZeptoJS.

@LFeh sim, sim. E aí tem alguns utilitários na web para ajudar nessa tarefa tipo o [Icomoon](https://icomoon.io/app) :smile:

**Comment by rickbenetti** [comment]
Como mencionei em outra [Grids com ou sem flexbox](https://github.com/frontendbr/forum/issues/13) estou testando o Bulma.io, mas nunca fui de utilizar, sempre odiei o Bootstrap, já cheguei a testar mas não gostei até hoje de nenhum.

Teve um período que criei meu prórprio framework em LESS, mas removi o repositório do github, tava velho e antiquado.

Já testei também o Semantic UI, Foundation (achei o mais chato de todos), o Uikit faz muito tempo que não uso.

Normalmente eu crio meu CSS do zero, usava o LESS por ter me habituado mais com ele, como me falta em muito a lógica ainda o SASS tem muita coisa legal, mas to penando pra usar rs, prefiro fazer do zero meus CSS's por ter só o que preciso evitando carregar  uma montanha de coisas desnecessárias.

**Comment by williamknn** [comment]
Eu geralmente uso Bootstrap porque nos projetos onde eu entro sempre a galera já tá trabalhando com Bootstrap, então não tenho como escapar. Mas já iniciei projeto do zero com Bootstrap também, só que nós personalizamos o estilo para deixar com um tema próprio. 

Recentemente eu descobri o FlatIcons que o @kevinthomas-dev mencionou, comecei a usar alguns ícones e tão logo quanto possível, irei remover o Font-Awesome do projeto no qual estou trabalhando.

Pré-processador eu sempre usei SASS.

**Comment by victorserpac** [comment]
Muito bom ter uma ideia geral de como estão achando soluções para trabalhar, irei dar uma estudada geral nos pontos que foram destacados e analisar a melhor alternativa pro meu projeto atual, até pq essa necessidade por mudar em algum outro trampo.

Ter uma noção das possibilidades foi demais, todo conteúdo aqui será muito útil. Valeu galera :D

**Comment by willycamargo** [comment]
@victorserpac , acho interessante renomear a issue para algo que relacione com CSS. Abri pensando que se tratava de Framework Javascript. Apenas por questão de semântica :smiley: 

Aqui o projeto segue a [Material Design Guideline](https://www.google.com/design/spec/material-design/introduction.html) e estamos utilizando o [MDL](http://www.getmdl.io/). Não é um framework tão maduro assim dizendo, então muita coisa que falta ser implementada no MDL eu tenho que implementar na unha (isso acontece bastante por ser um projeto SPA e com muitas funcionalidades).

**Comment by victorserpac** [comment]
@willycamargo boa, vou renomear aqui :D

Interessante o MDL, provável que tenha um futuro bem bom eu acredito.

**Comment by willycamargo** [comment]
Bem, é da própria comunidade do Google. Acredito ter um futuro promissor, deixando para trás outros frameworks que seguem a spec do Material Design.

**Comment by DevCarlosAugusto** [comment]
Só Frameworks foda, ultimamente eu tenho tentado construir um repositório "base" pra iniciar projetos.
Tenho pego ideias de algumas matérias para construção de Grids. talvez até o meio do ano tenha uma base.

Mas tenho utilizado mais grids soltos mesmo, tipo PureCSS, Miligram e coisas do tipo.

**Comment by brunoravanhani** [comment]
Eu gosto do Bootstrap pra quando preciso fazer algo rápido pra focar no back-end, mas geralmente utilizo somente o sistema de grids (geralmente o do Pure) e faço o resto na mão, mas gosto de fazer as coisas do meu jeito, fiquei com traumas de trabalhar com coisas tipo Primefaces que é totalmente engessado.

**Comment by felipefialho** [comment]
Percebo que muita gente utiliza frameworks CSS pro causa das soluções de grid e eu também fazia isso.

Recomendo fortemente vocês começarem a usar o [Jeet](http://jeet.gs/) ou o [Lost](https://github.com/peterramsing/lost) para esse fim.

**Comment by marquinhusgoncalves-zz** [comment]
Esse tópico esta sendo muito importante já que aqui na agência estamos querendo mudar já que usamos apenas o grid do Bootstrap, e o mais legal que o papo esta em alto nível e ninguém esta criticando um ou outro framework e sim mostrando os porques de usar um ou outro, acho isso legal porque tudo depende do projeto que esteja trabalhando, as pessoas que estão no time e o padrão da empresa, e isso que acho válido saber o que cada um usa e o porque. Muito obrigado galera...

**Comment by danilovaz** [comment]
@LFeh o Jeet não é mais uma boa usar, correto? vi que não estão mais dando manutenção nele. O Lost me parece que está indo pro mesmo caminho rs, vi que eles estavam montando o release 7.0.0 mas deram uma parada.

Eu estou começando a testar Lost + Rupture + Kout Swiss em um projeto grande, vamos ver o que acontece :stuck_out_tongue_closed_eyes:

**Comment by felipefialho** [comment]
@danilovaz Eu acho que ainda é uma boa, é um grid system bastante estável e com uma comunidade ativa por trás.

Mas aos poucos acredito que as pessoas vão mudar para outras opções. Eu mesmo estou testando Lost e gostando bastante, [podem ver aqui](https://github.com/frontendbr/eventos/tree/interface), uma das vantagens é usar flexbox bem facilmente.

**Comment by Cleysonlb** [comment]
Atualmente no meu workflow uso Jeet como sistema de grid e Rupture para media queries, uma solução que muitos utilizam e tem bastante conteúdo na web falando sobre, é simples, fácil, escalável.

**Comment by danilovaz** [comment]
eu fui perturbar  os caras do Lost no Twitter :stuck_out_tongue_closed_eyes: a respeito do Jeet e responderam isso:

```
@_danilovaz Jeet is no longer maintained as far as I know so Lost is indeed a better long-term solution{...}
```

de qualquer forma, acho que vou ficar com o Lost mesmo por enquanto, já que ele tende a continuar.
